core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -455,7 +459,18 @@ object ProducerStateManager {
 
   private def deleteSnapshotFiles(dir: File, predicate: Long => Boolean = _ => true) {
     listSnapshotFiles(dir).filter(file => predicate(offsetFromFile(file))).foreach { file =>
-      Files.deleteIfExists(file.toPath)
+      try{
+        logger.info(s"Sleep before deleteIfExists, thread Id: ${Thread.currentThread().getId}")
```

Comment:
Can you add the method name to the message. deleteIfExists is called from multiple places. This way we would now the logic where this is happening... like Sleep before deleting snapshot file "{file.toPath}", thread Id:...

Reply:
Ok.